### DISCUSSION TITLE
Add new shapes in devs namespace

### DISCUSSION THREAD

**Discussion by KarolMilewski** [clientIO/joint#1880] [Q&A] [answered]
Hi,
how can I add new shapes in devs namespace?
When using jointjs2.x I can add new shapes like `joint.shapes.devs.MyShape = joint.shapes.devs.Model.extend({})`. Now in jointjs.3x its works only in development. When I build the same code get some errors `MyShape is not exported by 'node_modules/jointjs/src/shapes/devs.mjs'`. 
I have the same problem when I use code from your example/demo code and change myNamespace to devs.
```js
    Object.assign(shapes, {
      devs: {
        MyShape,
      },
    })
```
Console output => `Uncaught (in promise) TypeError: Cannot set property devs of #<Object> which has only a getter`
I can't create new namespaces because I have tons of graphs saved with devs.MyShape type.

**Comment by kumilingus** [comment, marked as answer]
It probably does not like the fact that you are trying to override an imported module.

You can create a new namespace like this:
```js
const namespace = {
   devs: {
     ...joint.shapes.devs,
     MyShape
   },
   ...joint.shapes
}

const graph = new joint.dia.Graph({}, { cellNamespace: namespace });
const paper = new joint.dia.Paper({ cellViewNamespace: namespace, /* ... */ });
```

**Reply by KarolMilewski** [reply]
Thnx, but need change ordering in namespace :) 
eg. working code
```js
import { dia, shapes, util, V } from 'jointjs'

const namespace = {
   ...shapes,
   devs: {
     ...shapes.devs,
     MyShape
   },
}

const graph = new joint.dia.Graph({}, { cellNamespace: namespace });
const paper = new joint.dia.Paper({ cellViewNamespace: namespace, /* ... */ });

const newShape = new namespace.devs.MyShape({...})
 graph.addCell(newShape)
```

**Reply by kumilingus** [reply]
Of course, make sense :)
